### ISSUE TITLE
Updates to dev UI

### ISSUE THREAD

**Issue by gorkem** [jozu-ai/kitops#300]
**Describe the problem you're trying to solve**
Dev UI should be more helpful with helping the application developers integrate with the model. 

**Describe the solution you'd like**
- Generate/show example code as the parameters and prompts are entered. 
- A way to see JSON communicated between the server and responses
- hide or drop the list of preferences to highlight more frequently used ones
- Ability to see the README.md from the ModelKit if it has one

**Comment by annigro** [comment]
https://build.nvidia.com/explore/discover

**Comment by annigro** [comment]
[UX Design](https://www.figma.com/design/rk91xQQKvHZfcV3sqV4ReG/KitOps---dev-command?node-id=30%3A2699&t=KRFHr2Yb1aRTy4a9-1)

**Comment by gorkem** [comment]
Here is some more clarification on the requests.

`POST /completion` is an API specific to llama.cpp server. There is example code available for its usage which we will adjust for code generation feature.

`POST /v1/chat/completions` The endpoint for chat completion API is compatible with the OpenAI endpoint. We can use the existing OpenAPI libraries for code generation.

## Options

The initial thought was to reduce the options to match OpenAI. However, with further thinking and considering the options are supported by both endpoints. We should continue to use all options but categorize them better.

### Categories

#### Text Generation Controls

- Temperature: Controls randomness.
- Top_k: Limits token selection to the most probable.
- Top_p: Limits token selection by cumulative probability.
- Min_p: Sets a minimum probability threshold.
- N_predict: Max tokens to predict. (This could be a category of its own)

#### Sampling and Diversity

- Tfs_z: Tail-free sampling to control diversity.
- Typical_p: Locally typical sampling for natural text.
- Presence_penalty, frequency_penalty: Adjusts likelihood based on token presence and frequency.
- Mirostat, mirostat_tau, mirostat_eta: Controls complexity and randomness.
- Repeat_penalty, repeat_last_n: Controls repetition.

#### Advanced Settings and Customization

- Grammar, json_schema: Enables grammar-based sampling with JSON schema.
- api_key

#### Probability and Statistical Controls

- N_probs: Provides probabilities for top N tokens.
- Min_keep: Ensures a minimum number of tokens are returned.

## single or multiple page

I have not really found a good reason to keep the multiple page. It is unintuitive to go back and forth for changing the parameters. I suggest we do a single page implementation.

**Comment by annigro** [comment]
--> We talked about how the generated code for chat mode can get really long and therefore messy. @gorkem Have you come up with a solution?

Also, I do not see this problem occurring on build.nvidia.com. 🧐

**Comment by javisperez** [comment]
@annigro no, Gorkem's first two lines are related to internal code usage, and should be transparent for the UI other than one line. We talked about doing something like this for it:
```
message: [{ actor: 'user', content: 'foo bar fooz' }]
```

and when is too long:
```
message: [{ actor: 'user', content: 'foo bar ...' }] <-- ellipsis but for the message's content only, and only if is too long.
```

but clicking on "copy code" would always copy the whole thing, regardless of the ellipsis.

**Comment by annigro** [comment]
@gorkem  [Figma file to review](https://www.figma.com/design/rk91xQQKvHZfcV3sqV4ReG/KitOps---dev-command?node-id=230-3153&t=pQ6T5ZrcmwhXgbGU-1)

**Comment by gorkem** [comment]
I did a pass left a few comments.

**Comment by annigro** [comment]
[Categories and values for devmode](https://gist.github.com/gorkem/4e16834cd402f06a0a60eae1957423d3)

**Comment by javisperez** [comment]
This has been merged to main but will be "in prod" in the next Kit Release
